File: php/features.php
<?php
declare(strict_types=1);

// MockApiService SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MockApiServiceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MockApiServiceBaseFeature();
            case "test":
                return new MockApiServiceTestFeature();
            default:
                return new MockApiServiceBaseFeature();
        }
    }
}

File: py/features.py
# MockApiService SDK feature factory

from feature.base_feature import MockApiServiceBaseFeature
from feature.test_feature import MockApiServiceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MockApiServiceBaseFeature(),
        "test": lambda: MockApiServiceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
